ios/Sources/Component/InlineTitleNavigationBarScrollView.swift

```diff
@@ -0,0 +1,54 @@
+import SwiftUI
+import Introspect
+
+private struct ContentOffsetPreferenceKey: PreferenceKey {
+    static var defaultValue: CGFloat = 0
+
+    static func reduce(value: inout CGFloat, nextValue: () -> CGFloat) {}
+}
+
+/// A vertical scroll view applies scrollEdgeAppearance when its content reaches the matching edge of the navigation bar
+public struct InlineTitleNavigationBarScrollView<Content: View>: View {
+
+    public var content: Content
+    public var showsIndicators: Bool
+    @State private var shouldApplyScrollEdgeAppearance = true
+
+    public init(
+        showsIndicators: Bool = true,
+        @ViewBuilder content: () -> Content
+    ) {
+        self.showsIndicators = showsIndicators
+        self.content = content()
+    }
+
+    public var body: some View {
+        let _ = shouldApplyScrollEdgeAppearance
+        // Reference: https://medium.com/@maxnatchanon/swiftui-how-to-get-content-offset-from-scrollview-5ce1f84603ec
+        GeometryReader { outsideProxy in
+            ScrollView(showsIndicators: showsIndicators) {
+                ZStack {
+                    GeometryReader { insideProxy in
+                        Color.clear
+                            .preference(
+                                key: ContentOffsetPreferenceKey.self,
+                                value: outsideProxy.frame(in: .global).minY - insideProxy.frame(in: .global).minY
+                            )
+                    }
+                    content
+                }
+            }
+            .onPreferenceChange(ContentOffsetPreferenceKey.self) { value in
+                shouldApplyScrollEdgeAppearance = value <= 0
+            }
+            .introspectViewController { viewController in
+                guard let navigationBar = viewController.navigationController?.navigationBar else { return }
+                let navigationItem = viewController.navigationItem
+                navigationItem.standardAppearance = shouldApplyScrollEdgeAppearance
+                    ? navigationBar.scrollEdgeAppearance
+                    : navigationBar.standardAppearance
+                navigationBar.layoutIfNeeded()
```

Comment:
Planed to change gently following pan gesture using `UIViewPropertyAnimator`, but there is no property to animate but change the appearance and `layoutIfNeeded()` which will lose the progress when exterior reasons case `navigarionBar` re-layout.

Reply:
> Planed to change gently following pan gesture

This is the default behavior when build with Xcode 13, so we should remove `InlineTitleNavigationBarScrollView` when migrating to Xcode 13.